src/state/slices/limitOrderSlice/limitOrderSlice.ts

```diff
@@ -0,0 +1,211 @@
+import type { PayloadAction } from '@reduxjs/toolkit'
+import { createSlice } from '@reduxjs/toolkit'
+import type { QuoteId } from '@shapeshiftoss/types/dist/cowSwap'
+import assert from 'assert'
+import type { InterpolationOptions } from 'node-polyglot'
+
+import { TransactionExecutionState } from '../tradeQuoteSlice/types'
+import {
+  initialState,
+  limitOrderSubmissionInitialState,
+  LimitOrderSubmissionState,
+} from './constants'
+import { buildUnsignedLimitOrder } from './helpers'
+import type { LimitOrderActiveQuote } from './types'
+
+export const limitOrderSlice = createSlice({
+  name: 'limitOrder',
+  initialState,
+  reducers: {
+    clear: state => ({
+      ...initialState,
+      orderSubmission: state.orderSubmission, // Leave the limit order submission state alone
+    }),
+    setActiveQuote: (state, action: PayloadAction<LimitOrderActiveQuote | undefined>) => {
+      if (action.payload === undefined) {
+        state.activeQuote = undefined
+        return
+      }
+      const quoteId = action.payload.response.id
+
+      if (!quoteId) {
+        console.error('Missing quoteId')
+        return
+      }
+      state.activeQuote = action.payload
+      state.orderSubmission[quoteId] = limitOrderSubmissionInitialState
+    },
+    setLimitOrderInitialized: (state, action: PayloadAction<QuoteId>) => {
+      state.orderSubmission[action.payload].state = LimitOrderSubmissionState.Previewing
+    },
+    confirmSubmit: (state, action: PayloadAction<QuoteId>) => {
+      const limitOrderQuoteId = action.payload
+
+      // If there is no active quote, the state is corrupt.
+      if (state.activeQuote === undefined) {
+        console.error('Attempted to confirm an non-existent limit order quote')
+        return
```

Comment:
Also from reading the view-layer, looks like there are two layers (paranoia early return in submit fn and early render null) which will block this from happening.
Since we already have runtime checks against this, can we narrow down `state.activeQuote` / non-null assert it to be defined here? 
Or just go with the good ol' "this shouldn't happen but TS mang" comment